pkg/transport/setup/rpc.go

```diff
@@ -0,0 +1,71 @@
+package setup
+
+import (
+	"context"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/skycoin/dmsg/cipher"
+	"github.com/skycoin/skycoin/src/util/logging"
+
+	"github.com/skycoin/skywire/pkg/transport"
+)
+
+// TransportGateway that exposes methods to be used via RPC
+type TransportGateway struct {
+	tm  *transport.Manager
+	log *logging.Logger
+}
+
+// TransportRequest to perform an action over RPC
+type TransportRequest struct {
+	RemotePK cipher.PubKey
+	Type     string
+}
+
+// UUIDRequest contains id in UUID format
+type UUIDRequest struct {
+	ID uuid.UUID
+}
+
+// TransportResponse specifies an existing transport to remote node
+type TransportResponse struct {
+	ID     uuid.UUID
+	Local  cipher.PubKey
+	Remote cipher.PubKey
+	Type   string
+	IsUp   bool
+}
+
+// BoolResponse is a simple boolean wrapped in structure for RPC responses
+type BoolResponse struct {
+	Result bool
+}
+
+// AddTransport specified by request
+func (gw *TransportGateway) AddTransport(req TransportRequest, res *TransportResponse) error {
+	ctx := context.Background()
+	gw.log.WithField("PK", req.RemotePK).WithField("type", req.Type).Info("Adding transport")
+	mt, err := gw.tm.SaveTransport(ctx, req.RemotePK, req.Type, transport.LabelSkycoin)
```

Comment:
Logic itself is fine. But I believe this does not return an error if the transport could not be setup or is not up but has been dialed. We probably need to update the logic of this function itself.